hail/python/hailtop/utils/utils.py

```diff
@@ -326,6 +326,15 @@ async def __aexit__(self,
         self.release()
 
 
+async def cancel_task(t):
+    assert not t.done()
```

Comment:
So I think this will potentially fail since you separated the test for done and this check with a coroutine invocation.  Between the call to `cancel_task(t)` and this assert, the task `t` could get scheduled and finish.  I think you can move the test for done into `cancel_task`, or just remove this check.